### PR TITLE
39_retry_after

### LINKED ISSUE #39: Propagate Retry-After header value
### Motivation

When I disable the default resilience logic in the `ContentManagementClient` and I exceed the [API rate limit](https://developer.kenticocloud.com/v1/reference#content-management-api-v2-rate-limits), I have no way to find out how long I should wait before retrying the request.
### Proposed solution

I should be able to get the value of the `Retry-After` header in the `ContentManagementException`, e.g., in the [JavaScript Content Management SDK](https://github.com/Kentico/kentico-cloud-js/tree/b39faf662544ae88ddcc698c672be91cf44aa073/packages/content-management), the [original error is included](https://github.com/Kentico/kentico-cloud-js/blob/b39faf662544ae88ddcc698c672be91cf44aa073/packages/content-management/lib/models/shared/shared-models.ts#L34).

### PR DESCRIPTION
**Pull request by gormal**
### Motivation

Fixes #39

If a response with status 429 (too many requests) is received default retry policy will honor the retry-after header.

### CODE REVIEW

**Review by Simply007 — COMMENTED** [review]
We should try to be consistent with https://github.com/Kentico/kontent-delivery-sdk-net/blob/master/Kentico.Kontent.Delivery/RetryPolicy/DefaultRetryPolicy.cs

We should have coherent in these resources:
* https://github.com/Kentico/kontent-delivery-sdk-net/wiki/Retry-capabilities
* https://kontent.ai/learn/reference/delivery-api/?search=retry+policy#section/Errors/HTTP-status-codes-summary

**Comment by gormal** [comment]
> We should try to be consistent with https://github.com/Kentico/kontent-delivery-sdk-net/blob/master/Kentico.Kontent.Delivery/RetryPolicy/DefaultRetryPolicy.cs
> 
> We should have coherent in these resources:
> 
> * https://github.com/Kentico/kontent-delivery-sdk-net/wiki/Retry-capabilities
> * https://kontent.ai/learn/reference/delivery-api/?search=retry+policy#section/Errors/HTTP-status-codes-summary

I've just had a look and it seems like we can straight up copy it.

File: Kentico.Kontent.Management/Modules/ResiliencePolicy/DefaultResiliencePolicyProvider.cs [hunk 1]
```diff
@@ -10,6 +14,22 @@ namespace Kentico.Kontent.Management.Modules.ResiliencePolicy;
...
+            HttpStatusCode.InternalServerError,
+            HttpStatusCode.BadGateway,
+            HttpStatusCode.ServiceUnavailable,
+            HttpStatusCode.GatewayTimeout,
+    };
+    
+    private static readonly HttpStatusCode[] StatusCodesWithPossibleRetryHeader =
+    {
+            (HttpStatusCode)429, // Too Many Requests
+            HttpStatusCode.ServiceUnavailable,
```
Comment (Simply007) [inline Kentico.Kontent.Management/Modules/ResiliencePolicy/DefaultResiliencePolicyProvider.cs:30]: (I know it is also in DAPI SDK)
Is it valid that this error (503) can also return retry-after? - It does not seems like ti in https://kontent.ai/learn/reference/management-api-v2/#section/Errors/HTTP-status-codes-summary It

**Review by Simply007 — APPROVED** [review]
Just one remark - which might be more focused on Docs. What do you think @jancerman

File: Kentico.Kontent.Management/Modules/ResiliencePolicy/DefaultResiliencePolicyProvider.cs (hunk 1 above)
Comment (jancerman) [reply Kentico.Kontent.Management/Modules/ResiliencePolicy/DefaultResiliencePolicyProvider.cs:30]: I think it's unusual to return a `retry-after` header for any response other than 429. 5xx tells me it's an internal server error, so I wouldn't expect any extra information.
However, [MDN page on the `retry-after` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) states that this usage "indicates how long the service is expected to be unavailable." Is this what we're signaling with the header? If so, I guess it's okay. If not, I wouldn't include it.

**Pull request merged**